src/views/viewer/ChannelView/ChannelView.tsx

```diff
@@ -83,15 +140,61 @@ export const ChannelView: React.FC = () => {
             )}
           </TitleContainer>
           <StyledButtonContainer>
-            <Button variant={isFollowing ? 'secondary' : 'primary'} onClick={handleFollow} size="large">
+            <StyledButton
+              icon={isFollowing ? <SvgGlyphCheck /> : <SvgGlyphPlus />}
+              variant={isFollowing ? 'secondary' : 'primary'}
+              onClick={handleFollow}
+              size="large"
+            >
               {isFollowing ? 'Unfollow' : 'Follow'}
-            </Button>
+            </StyledButton>
           </StyledButtonContainer>
         </TitleSection>
-      </Header>
-      <VideoSection className={transitions.names.slide}>
-        <InfiniteVideoGrid channelId={id} showChannel={false} />
-      </VideoSection>
+        <TabsContainer>
+          <Tabs initialIndex={0} tabs={mappedTabs} onSelectTab={handleSetCurrentTab} />
+          {currentTabName === 'Videos' && (
+            <SortContainer>
+              <Text variant="body2">Sort by</Text>
+              <Select helperText={null} value={sortVideosBy} items={SORT_OPTIONS} onChange={handleSorting} />
+            </SortContainer>
+          )}
+        </TabsContainer>
+        {(() => {
+          switch (currentTabName) {
+            case 'Videos':
+              return (
+                <>
+                  <VideoSection className={transitions.names.slide}>
+                    <Grid maxColumns={null} onResize={handleOnResizeGrid}>
+                      {videosWithPlaceholders.map((video, idx) => (
+                        <VideoPreview key={idx} id={video.id} showChannel={false} />
+                      ))}
+                    </Grid>
+                  </VideoSection>
+                  <PaginationContainer>
+                    <Pagination
+                      onChangePage={handleChangePage}
+                      page={currentPage}
+                      itemsPerPage={videosPerPage}
+                      totalCount={totalCount}
+                    />
+                  </PaginationContainer>
+                </>
+              )
+            case 'About':
+              return <ChannelAbout />
+          }
+        })()}
+      </LimitedWidthContainer>
     </ViewWrapper>
   )
 }
+
+const usePagination = (currentTab: number) => {
+  const [currentPage, setCurrentPage] = useState(0)
+  // reset the pagination when changing tabs
+  useEffect(() => {
+    setCurrentPage(0)
+  }, [currentTab])
+  return { currentPage, setCurrentPage }
+}
```

Comment:
Not sure what's causing it, I tried to investigate this, but when I go to the channel from `/channels` I often see this: 
![image](https://user-images.githubusercontent.com/51168865/125963021-d3227ab5-01db-4942-b216-e6380d8ac133.png)
For some reason videos not loading correctly.  This is weird because it doesn't happen on /studio/channel